icinga2/icinga.go

```diff
@@ -141,6 +161,40 @@ func (s *WebClient) FilteredQuery(url string, filter QueryFilter, result, errmsg
 	return s.napping.Send(&req)
 }
 
+func (s *WebClient) SetIcingaUrl(url string) {
+	s.URL = url
+}
+
+func (s *MockClient) SetIcingaUrl(url string) {
+	conf := s.GetClientConfig()
+	conf.SetIcingaUrl(url)
+}
+
+func (s *WebClient) TestIcingaApi(url string) error {
+	var results, errmsg Results
+
+	resp, err := s.napping.Get(url+"/v1", nil, &results, &errmsg)
+	if err != nil {
+		return err
+	}
+
+	if resp.HttpResponse().StatusCode != 200 {
+		return err
+	}
+
+	return nil
+}
+
+func (s *MockClient) TestIcingaApi(url string) error {
+	session := s.GetClientConfig().napping
+
+	_, err := session.Get(url+"/v1", nil, nil, nil)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
It makes no sense that the MockClient tries to make a HTTP request using a napping client which wasn't initialized.

Instead I'd do something like the following here:

```suggestion
	url, err := url.Parse(url)
	if err != nil {
		return err
	}
	if url.Host == "" {
		return fmt.Errorf("URL without hostname not supported: %v", url)
	}
	return nil
```

which just parses the URL and verifies that the method is called with a URL which includes a hostname.